ext/hook/uhook_legacy.c

```diff
@@ -306,36 +306,27 @@ static void dd_uhook(INTERNAL_FUNCTION_PARAMETERS, bool tracing, bool method) {
     zval *prehook = NULL, *posthook = NULL, *config_array = NULL;
     bool run_when_limited = false, allow_recursion = false;
 
-    ZEND_PARSE_PARAMETERS_START_EX(ZEND_PARSE_PARAMS_QUIET, 1 + method, 2 + method + !tracing)
+    ZEND_PARSE_PARAMETERS_START(1 + method, 2 + method + !tracing)
         // clang-format off
         if (method) {
             Z_PARAM_STR(class_name)
         }
         Z_PARAM_STR(method_name)
         Z_PARAM_OPTIONAL
-        if (!tracing) {
+
+        if (_i >= 0 && (uint32_t)_i == ZEND_NUM_ARGS()) {
+            break;
+        }
+
+        zval *_current_arg = _arg + _i - method;
```

Comment:
I think that's wrong?
```suggestion
        zval *_current_arg = _arg + _i;
```

Reply:
Long story short: Yes, it was, but the `_i - method` would always be `1`, so it appeared to have fixed my issue in the first place, while this was just luck. I didn't initially realize `_arg` was being incremented on the fly.

Therefore, I believe `_arg + 1` should represent the current arg. At this point in the code, `_arg` is a string (the method name).

The check done right before is used because, at this point, the args are optional (although I don't understand why it makes any sense to be able to set up a hook without any callbacks associated with it...). Therefore, it mayn't make sense to check for the current arg (since it doesn't exist).

Honestly, I wouldn't say I like this code section, and I find it pretty ugly. If it works, I guess that's what ultimately matters, but it doesn't feel idiomatic at all :/